test/m4/appframework/appframework_test.go

```diff
@@ -2066,7 +2066,207 @@ var _ = Describe("m4appfw test", func() {
+
+			// Set AppsRepoPollInterval for Cluster Master to 120s
+			testcaseEnvInst.Log.Info("Set AppsRepoPollInterval for Cluster Master to 120s")
+			cm.Spec.AppFrameworkConfig.AppsRepoPollInterval = int64(120)
+			err = deployment.UpdateCR(ctx, cm)
+			Expect(err).To(Succeed(), "Failed to change AppsRepoPollInterval value for Cluster Master")
+
+			// Get instance of current Search Head Cluster CR with latest config
+			shc = &enterpriseApi.SearchHeadCluster{}
+			err = deployment.GetInstance(ctx, deployment.GetName()+"-shc", shc)
+			Expect(err).To(Succeed(), "Failed to edit  Search Head Cluster")
+
+			// Set AppsRepoPollInterval for Search Head Cluster to 120s
+			testcaseEnvInst.Log.Info("Set AppsRepoPollInterval for Search Head Cluster to 120s")
+			shc.Spec.AppFrameworkConfig.AppsRepoPollInterval = int64(120)
+			err = deployment.UpdateCR(ctx, shc)
+			Expect(err).To(Succeed(), "Failed to change AppsRepoPollInterval value for Search Head Cluster")
+
+			// Change status to 'ON' in config map for Cluster Master and Search Head Cluster
+			testcaseEnvInst.Log.Info("Change status to 'ON' in config map for Cluster Master")
+			config, err = testenv.GetAppframeworkManualUpdateConfigMap(ctx, deployment, testcaseEnvInst.GetName())
+			Expect(err).To(Succeed(), "Unable to get config map")
+
+			config.Data["ClusterMaster"] = strings.Replace(config.Data["ClusterMaster"], "off", "on", 1)
+			err = deployment.UpdateCR(ctx, config)
+			Expect(err).To(Succeed(), "Unable to update config map for Cluster Master")
+
+			testcaseEnvInst.Log.Info("Change status to 'ON' in config map for Search Head Cluster")
+			config.Data["SearchHeadCluster"] = strings.Replace(config.Data["SearchHeadCluster"], "off", "on", 1)
+			err = deployment.UpdateCR(ctx, config)
+			Expect(err).To(Succeed(), "Unable to update config map for Search Head Cluster")
+
+			// Verify status is 'ON' in config map for Cluster Master and Search Head Cluster
+			testcaseEnvInst.Log.Info("Verify status is 'ON' in config map for Cluster Master and Search Head Cluster")
+			config, _ = testenv.GetAppframeworkManualUpdateConfigMap(ctx, deployment, testcaseEnvInst.GetName())
+			Expect(strings.Contains(config.Data["ClusterMaster"], "status: on") && strings.Contains(config.Data["SearchHeadCluster"], "status: on")).To(Equal(true), "Config map update not complete")
+
+			// Ensure that the Cluster Manager goes to Ready phase
+			testenv.ClusterManagerReady(ctx, deployment, testcaseEnvInst)
+
+			// Ensure the Indexers of all sites go to Ready phase
+			testenv.IndexersReady(ctx, deployment, testcaseEnvInst, siteCount)
 
+			// Ensure Indexer cluster configured as multisite
+			testenv.IndexerClusterMultisiteStatus(ctx, deployment, testcaseEnvInst, siteCount)
+
+			// Ensure Search Head Cluster go to Ready phase
+			testenv.SearchHeadClusterReady(ctx, deployment, testcaseEnvInst)
+
+			// Get Pod age to check for pod resets later
+			splunkPodAge = testenv.GetPodsStartTime(testcaseEnvInst.GetName())
+
```

Comment:
The app upload for v2 should be before this.

And then we should be checking at this point that the apps are not upgraded even though we modified config map to "on". 

The point of the test case would be that since `AppsRepoPollInterval` is set to non-zero. Modifying the configmap would not trigger an app poll